spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestStoragePartitionedJoinsInRowLevelOperations.java

```diff
@@ -257,14 +261,22 @@ private void checkMerge(RowLevelOperationMode mode) {
           SparkPlan plan =
               executeAndKeepPlan(
                   "MERGE INTO %s AS t USING %s AS s "
-                      + "ON t.id = s.id AND t.dep = s.dep AND t.dep = 'hr'"
+                      + "ON t.id = s.id AND t.dep = s.dep "
                       + "WHEN MATCHED THEN "
                       + "  UPDATE SET t.salary = s.salary "
                       + "WHEN NOT MATCHED THEN "
                       + "  INSERT *",
                   tableName, tableName(OTHER_TABLE_NAME));
           String planAsString = plan.toString();
-          Assert.assertFalse("Should be no shuffles with SPJ", planAsString.contains("Exchange"));
+          if (mode == COPY_ON_WRITE) {
+            int actualNumShuffles = StringUtils.countMatches(planAsString, "Exchange");
+            Assert.assertEquals("Should be 1 shuffle with SPJ", 1, actualNumShuffles);
+            Assert.assertTrue(
```

Comment:
nit: using `Assertions.assertThat(planAsString).contains(...)` / `Assertions.assertThat(planAsString).doesNotContain(...)` will actually give you the entire `planAsString` in case the assertion ever fails (which makes debugging much easier)